python/pyspark/pandas/sql_formatter.py

```
@@ -201,8 +203,7 @@ def sql(
     try:
         sdf = session.sql(formatter.format(query, **kwargs), args)
     finally:
-        if not is_remote():
```

Can we throw an exception instead?

---

Sounds good. Updated